src/Associable/Shuffle.php

```diff
@@ -24,9 +24,16 @@ public function __construct(Associable $associable)
     /** @return array<TKey,TValue> */
     public function toAssocArray(): array
     {
-        $items = $this->associable->toAssocArray();
-        shuffle($items);
+        $old = $this->associable->toAssocArray();
+        $new = [];
 
-        return $items;
+        $keys = array_keys($old);
+        shuffle($keys);
+
+        foreach ($keys as $key) {
+            $new[$key] = $old[$key];
+        }
+
+        return $new;
```

Comment:
I'm not sure it's a good idea, not always native PHP implementation is wrong ;)

In many languages maps are not sortable, it's PHP thing that map and list is the same type.
E.g. when you save JSON object to MySQL and read it back, JSON object properties can have different order, probably due some optimization, maybe they didn't think someone will care :P

That's why I think that shuffling assoc without changing key-value association is just doing nothing :)

Reply:
NVM, I understood what was the problem, that `shuffle` function actually turns assoc into list :disappointed: